Looks like when I run `bazel test tensorflow/tools/api/tests:api_compatibility_test`, it fails similar to github:
```
ERROR:tensorflow:1 differences found between API and golden.
E0122 01:03:45.536555 139929196537664 api_compatibility_test.py:264] 1 differences found between API and golden.
Issue 1 : New object tensorflow.experimental.tensorrt.ConversionParams found (added).
[  FAILED  ] ApiCompatibilityTest.testAPIBackwardsCompatibility
...
ERROR:tensorflow:1 differences found between API and golden.
E0122 01:03:54.671969 139929196537664 api_compatibility_test.py:264] 1 differences found between API and golden.
Issue 1 : New object tensorflow.experimental.tensorrt.ConversionParams found (added).
[  FAILED  ] ApiCompatibilityTest.testAPIBackwardsCompatibilityV2
[ RUN      ] ApiCompatibilityTest.testNoSubclassOfMessage
[       OK ] ApiCompatibilityTest.testNoSubclassOfMessage
[ RUN      ] ApiCompatibilityTest.testNoSubclassOfMessageV1
[       OK ] ApiCompatibilityTest.testNoSubclassOfMessageV1
[ RUN      ] ApiCompatibilityTest.testNoSubclassOfMessageV2
[       OK ] ApiCompatibilityTest.testNoSubclassOfMessageV2
[ RUN      ] ApiCompatibilityTest.test_session
[  SKIPPED ] ApiCompatibilityTest.test_session
======================================================================
FAIL: testAPIBackwardsCompatibility (__main__.ApiCompatibilityTest)
testAPIBackwardsCompatibility (__main__.ApiCompatibilityTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/.cache/bazel/_bazel_root/233a431e54f1cbffefcb6321279dfb61/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/tools/api/tests/api_compatibility_test.runfiles/org_tensorflow/tensorflow/tools/api/
tests/api_compatibility_test.py", line 390, in testAPIBackwardsCompatibility
    omit_golden_symbols_map=omit_golden_symbols_map)
  File "/root/.cache/bazel/_bazel_root/233a431e54f1cbffefcb6321279dfb61/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/tools/api/tests/api_compatibility_test.runfiles/org_tensorflow/tensorflow/tools/api/
tests/api_compatibility_test.py", line 367, in _checkBackwardsCompatibility
    api_version=api_version)
  File "/root/.cache/bazel/_bazel_root/233a431e54f1cbffefcb6321279dfb61/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/tools/api/tests/api_compatibility_test.runfiles/org_tensorflow/tensorflow/tools/api/
tests/api_compatibility_test.py", line 287, in _AssertProtoDictEquals
    self.fail('%d differences found between API and golden.' % diff_count)
  File "/root/.cache/bazel/_bazel_root/233a431e54f1cbffefcb6321279dfb61/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/tools/api/tests/api_compatibility_test.runfiles/absl_py/absl/testing/absltest.py", l
ine 1655, in fail
    return super(TestCase, self).fail(self._formatMessage(prefix, msg))
AssertionError: 1 differences found between API and golden.
```

But when I run `bazel test tensorflow/tools/api/tests:api_compatibility_test -- --update_goldens True`, although the log reports the same failure, but it reports the final result as OK.

```
INFO:tensorflow:No differences found between API and golden.
I0117 18:57:16.801649 140141113898816 api_compatibility_test.py:290] No differences found between API and golden.
[       OK ] ApiCompatibilityTest.testAPIBackwardsCompatibilityV1
[ RUN      ] ApiCompatibilityTest.testAPIBackwardsCompatibilityV2
INFO:tensorflow:No differences found between API and golden.
I0117 18:57:19.938147 140141113898816 api_compatibility_test.py:290] No differences found between API and golden.
[       OK ] ApiCompatibilityTest.testAPIBackwardsCompatibilityV2
[ RUN      ] ApiCompatibilityTest.testNoSubclassOfMessage
[       OK ] ApiCompatibilityTest.testNoSubclassOfMessage
[ RUN      ] ApiCompatibilityTest.testNoSubclassOfMessageV1
[       OK ] ApiCompatibilityTest.testNoSubclassOfMessageV1
[ RUN      ] ApiCompatibilityTest.testNoSubclassOfMessageV2
[       OK ] ApiCompatibilityTest.testNoSubclassOfMessageV2
[ RUN      ] ApiCompatibilityTest.test_session
[  SKIPPED ] ApiCompatibilityTest.test_session
----------------------------------------------------------------------
Ran 7 tests in 11.279s

OK (skipped=1)
```

- Issue 1) When I run with `-- --update_goldens True`, it shows OK at the end although there are failures in the log.
- Issue 2) When I run with `-- --update_goldens True`, it doesn't update the golden.
- Question 1) Should I update the golden manually, given that the script doesn't work?
- Question 2) Should the manual update be adding a new pbtxt file given that we have exported a new object called `ConversionParams`?
- Question 3) What should be the name of the new pbtxt file for `ConversionParams`. Perhaps: 
`tensorflow/tools/api/golden/v2/tensorflow.experimental.tensorrt.-conversion-params.pbtxt`?

--- a/tensorflow/python/compiler/tensorrt/trt_convert.py
+++ b/tensorflow/python/compiler/tensorrt/trt_convert.py
@@ -986,7 +986,7 @@ class TrtGraphConverterV2(object):
                input_saved_model_dir=None,
                input_saved_model_tags=None,
                input_saved_model_signature_key=None,
-               conversion_params=TrtConversionParams()):
+               conversion_params=None):
     """Initialize the converter.
 
     Args:
@@ -1001,6 +1001,8 @@ class TrtGraphConverterV2(object):
       ValueError: if the combination of the parameters is invalid.
     """
     assert context.executing_eagerly()
+    if conversion_params is None:
+      conversion_params = TrtConversionParams()
     _check_trt_version_compatibility()
     _check_conversion_params(conversion_params, is_v2=True)
 

--- a/tensorflow/python/compiler/tensorrt/trt_convert.py
+++ b/tensorflow/python/compiler/tensorrt/trt_convert.py
@@ -159,14 +159,28 @@ class TrtConversionParams(object):
       max_batch_size: max size for the input batch. This parameter is only
         effective when is_dynamic_op=False which is not supported in TF 2.0.
     """
-    self.rewriter_config_template = rewriter_config_template
-    self.max_workspace_size_bytes = max_workspace_size_bytes
-    self.precision_mode = precision_mode
-    self.minimum_segment_size = minimum_segment_size
-    self.is_dynamic_op = is_dynamic_op
-    self.maximum_cached_engines = maximum_cached_engines
-    self.use_calibration = use_calibration
-    self.max_batch_size = max_batch_size
+    super(TrtConversionParams, self).__setattr__(
+        "rewriter_config_template", rewriter_config_template)
+    super(TrtConversionParams, self).__setattr__(
+        "max_workspace_size_bytes", max_workspace_size_bytes)
+    super(TrtConversionParams, self).__setattr__(
+        "precision_mode", precision_mode)
+    super(TrtConversionParams, self).__setattr__(
+        "minimum_segment_size", minimum_segment_size)
+    super(TrtConversionParams, self).__setattr__(
+        "is_dynamic_op", is_dynamic_op)
+    super(TrtConversionParams, self).__setattr__(
+        "maximum_cached_engines", maximum_cached_engines)
+    super(TrtConversionParams, self).__setattr__(
+        "use_calibration", use_calibration)
+    super(TrtConversionParams, self).__setattr__(
+        "max_batch_size", max_batch_size)
+
+  def __setattr__(self, key, value):
+    """Override __setattr__ to make class immutable."""
+    raise AttributeError(
+        "{} is immutable and thus cannot change its attributes".format(
+            self.__class__))
 
   def _replace(self,
                rewriter_config_template=None,
@@ -184,10 +198,11 @@ class TrtConversionParams(object):
     set parameters:
       DEFAULT_TRT_CONVERSION_PARAMS._replace(...)
     """
-    trt_conversion_params = TrtConversionParams()
+    conv_params_dict = {}
     for k, v in vars().items():
-      if v and (k != "self"):
-        setattr(trt_conversion_params, k, v)
+      if (v != None) and (k not in ("self", "conv_params_dict")):
+        conv_params_dict[k] = v
+    trt_conversion_params = TrtConversionParams(**conv_params_dict)
     return trt_conversion_params
 
   def __repr__(self):

--- a/tensorflow/tools/api/golden/v2/tensorflow.experimental.tensorrt.-conversion-params.pbtxt
+++ b/tensorflow/tools/api/golden/v2/tensorflow.experimental.tensorrt.-conversion-params.pbtxt
@@ -1,0 +1,9 @@
+path: "tensorflow.experimental.tensorrt.ConversionParams"
+tf_class {
+  is_instance: "<class \'tensorflow.python.compiler.tensorrt.trt_convert.TrtConversionParams\'>"
+  is_instance: "<type \'object\'>"
+  member_method {
+    name: "__init__"
+    argspec: "args=[\'self\', \'rewriter_config_template\', \'max_workspace_size_bytes\', \'precision_mode\', \'minimum_segment_size\', \'is_dynamic_op\', \'maximum_cached_engines\', \'use_calibration\', \'max_batch_size\'], varargs=None, keywords=None, defaults=[\'None\', \'1073741824\', \'FP32\', \'3\', \'True\', \'1\', \'True\', \'1\'], "
+  }
+}

